crates/binstalk/src/fetchers.rs

```diff
@@ -107,9 +127,75 @@ impl Data {
     }
 }
 
+impl RepoInfo {
+    /// If `repo` contains a subcrate, then extracts that and return it as
+    /// `{subcrate}%2F` and removes that subcrate path from `repo` to match
+    /// `scheme:/{repo_owner}/{repo_name}`
+    fn detect_subcrate(repo: &mut Url, repository_host: RepositoryHost) -> Option<String> {
+        if repository_host != RepositoryHost::GitHub {
```

Comment:
This PR only adds subcrate detection for GitHub for now.
We could create a new issue for other hosting services.

Reply:
I've added subcrate detection for GitLab.

Reply:
I've decided not to add subcrate detection for sourceforge, as it seems a lot of projects simply use it to provide pre-built artifacts.